src/etl/impl/LedgerFetcher.h

```diff
@@ -89,9 +89,15 @@ class LedgerFetcher {
     {
         LOG(log_.debug()) << "Attempting to fetch ledger with sequence = " << sequence;
 
-        auto response = loadBalancer_->fetchLedger(
-            sequence, true, !backend_->cache().isFull() || backend_->cache().latestLedgerSequence() >= sequence
-        );
+        auto const isCacheFull = backend_->cache().isFull();
+        auto const isLedgerCached = backend_->cache().latestLedgerSequence() >= sequence;
```

Comment:
Hmm, the naming of this variable is a bit weird. What this is testing is whether or not the object neighbors can be correctly computed for the desired ledger with the current state of the cache. That computation can only be done correctly if the cached ledger is one less than the desired ledger. So this is actually testing if the ledger is **_not_** cached.

The original computation is a bit weird (probably my fault), because it should really be an inequality computation (`backend_->cache().latestLedgerSequence() != sequence - 1`). I feel like there is some hidden assumption that if the cache is full, then it can't be less than sequence - 1, so the logic amounts to the same thing. But I do feel it's confusing written like this. Double check me though, because I haven't looked at this code in a while. Wanted to chime in though, because I know this part of the code is tricky.

Reply:
Thanks for commenting it! Yes, we are not sure about this part, the second condition seems impossible to trigger.  Since it is not the root cause of this issue , we just put more log here to help us investigate. 
`backend_->cache().latestLedgerSequence() != sequence - 1` is not quite right too. Imagine if Clio has many ledgers to catch up, we will fetch these ledgers in parallel. The cache maybe not available for these ledgers when clio is fetching , but the cache will be available when these ledger are  deserialised (because deserialisition happens synchronously).  We don't wanna request more data than we need.

Reply:
I am going to resolve the conversation to get this merged. Please do comment in other thread.